wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java

```diff
@@ -421,16 +421,37 @@ public Page createPage(Wiki wiki, String parentPageName, Page page) throws WikiE
       permissions = this.getWikiDefaultPermissions(wiki.getType(), wiki.getOwner());
       page.setPermissions(permissions);
     }
-    //init permission for page's creator
+    // init permission for page's creator
     String currentUser = org.exoplatform.wiki.utils.Utils.getCurrentUser();
-    if(!isExistedPermissionEntry(currentUser, permissions, page)) {
-      PermissionEntry permissionEntry = new PermissionEntry(currentUser, "", IDType.USER, new Permission[] {
-                                                                  new Permission(PermissionType.VIEWPAGE, true),
-                                                                  new Permission(PermissionType.EDITPAGE, true),
-                                                                  new Permission(PermissionType.ADMINPAGE, true) 
-                                                                  });
-      permissions.add(permissionEntry);
-      page.setPermissions(permissions);
+    PermissionEntry currentUserPermEnty = null;
+    outerloop:
+    for (PermissionEntry permEntry : page.getPermissions()) {
+      // get user permission entry  
+      if (permEntry.getId().equals(currentUser)) {
+        currentUserPermEnty = permEntry;
+        for (Permission permission : permEntry.getPermissions()) {
+          if (permission.getPermissionType().equals(PermissionType.ADMINPAGE)) {
+            if (permission.isAllowed()) {
+             break outerloop;
+            } else {
+              // If user has ADMINPAGE permission but the isAllowed = false
+              permission.setAllowed(true);

```

Comment:
Should break the loop also in this case, no need to go further :
if (!permission.isAllowed()) {
  // Update permEntry here
}
break;